Feat: Add check member

HwanCh0i opened the pull request:
## 사용자 정보 확인하기

- 로그인 버튼 클릭시 생성되는 idToken을 decode 해서 해당 사용자의  정보가 DB에 있는지 확인하는 로직입니다.

hannadoolsetnet committed:
Feat : Register user information

hannadoolsetnet committed:
Feat: Treat CORS error

hannadoolsetnet committed:
Merge pull request #3 from code-bottari/CORS_error

Feat: Treat CORS error

Commit:
Refactor: Modify schema

Commit:
Refactor: Error handling

Commit:
Rename: Create constants directory

Commit:
Feat: Add check member

Commit:
Refactor: Modify error handling

imjaehoo commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+  const { idToken } = req.body;
+
+  try {
+    const decodeToken = await admin
```
decode하는 함수가 아니라 decode한 결과물이니까 decodedToken? 아니면 그냥 token?

HwanCh0i replied:
decodedToken으로 수정하겠습니다!

imjaehoo commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+const MESSAGE = require("../constants/messages");
+const ERROR = require("../constants/error");
+
+router.get("/check_member", async (req, res, next) => {
```
언더스코어는 잘 안 쓸 거예요

HwanCh0i replied:
관련 컨벤션 확인했고, `"-"` 으로 변경하겠습니다.

imjaehoo commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+    const { code } = error;
+
+    if (code === AUTH_ID_TOKEN_REVOKED) {
+      next(createError(422, INVALID_TOKEN));
+      return;
```
newline before return를 규칙으로 삼는 곳도 많습니다.

hannadoolsetnet commented on constants/messages.js:
```diff
@@ -0,0 +1,8 @@
+const MESSAGE = {
+  UNEXPECTED_ERROR: "예상치 못한 오류가 발상하였습니다.",
```
오타 수정 부탁드립니다!

```suggestion
  UNEXPECTED_ERROR: "예상치 못한 오류가 발생하였습니다.",
```

hannadoolsetnet commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+
+const User = require("../models/User");
+
+const MESSAGE = require("../constants/messages");
+const ERROR = require("../constants/error");
```
단수, 복수 통일해 주세요.

```suggestion
const MESSAGES = require("../constants/messages");
const ERRORS = require("../constants/errors");
```

hannadoolsetnet commented on constants/messages.js:
```diff
@@ -0,0 +1,8 @@
+  OK: "ok",
+};
+
+module.exports = MESSAGE;
```
`MESSAGE` -> `MESSAGES` 🙏🏻

hannadoolsetnet commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+const ERROR = require("../constants/error");
+
+router.get("/check_member", async (req, res, next) => {
+  const { EXPIRED_TOKEN, UNEXPECTED_ERROR, OK } = MESSAGE;
+  const { AUTH_ID_TOKEN_REVOKED, AUTH_ID_TOKEN_EXPIRED } = ERROR;
```
스타일 통일을 위해 다음과 같이 수정 부탁드립니다.

```suggestion
  const {
    EXPIRED_TOKEN,
    UNEXPECTED_ERROR,
    OK,
  } = MESSAGES;
  
  const {
    AUTH_ID_TOKEN_REVOKED,
    AUTH_ID_TOKEN_EXPIRED,
  } = ERRORS;
```

hannadoolsetnet commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+
+    const user = await User.findOne({ email });
+
+    const hasUserData = user === null ? false : true;
```
boolean 값을 반환하는 삼항 연산자의 사용은 지양하는 것이 좋습니다.

대신 다음과 같이 작성할 수 있습니다!

```suggestion
    const hasUserData = user !== null;
```

SeaBlueMaker commented on routes/users.js:
```diff
@@ -0,0 +1,57 @@
+
+    const user = await User.findOne({ email });
+
+    const hasUserData = user === null ? false : true;
```
삼항연산자로 bool 값을 반환하고 계신데, 다음과 같이 작성하면 좋을 것 같습니다!
```suggestion
    const hasUserData = !!user;
```

chlghks replied:
`const hasUserData = user !== null;`
이러한 방식은 어떨까요?

HwanCh0i replied:
토의한 것 과 같이 이번 경우에는 
`const hasUserData = user !== null;`
작성하겠습니다!

Commit:
Style: modify code style

  - use destructuring style
  - modify constants identifier from  singular to plural
  - newline before return statement
  - remove conditional operator
